### PR TITLE
bug fix snippetConfig name or source same like will lost yml

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of PR is this?
bugfix

#### What this PR does / why we need it:

bug fix snippetConfig name or source same like will lost yml

### CODE REVIEW

File: modules/pipeline/services/pipelinesvc/snippet.go [hunk 1]
```diff
@@ -15,91 +15,63 @@ package pipelinesvc
...
-			yml, err := s.queryPipelineYAMLBySnippetConfig(&sc)
-			if err != nil {
-				return nil, err
-			}
-			if _, ok := yamlResults[sc.Source]; !ok {
-				yamlResults[sc.Source] = make(map[string]string)
-			}
-			yamlResults[sc.Source][sc.Name] = yml
+	for _, singleConfig := range singleSourceSnippetConfigs {
+		yml, err := s.queryPipelineYAMLBySnippetConfig(&singleConfig)
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/snippet.go:56]: async handle

File: modules/pipeline/services/pipelinesvc/snippet.go [hunk 2]
```diff
@@ -15,91 +15,63 @@ package pipelinesvc
...
+		if err != nil {
+			return nil, err
 		}
+		yamlResults[jsonparse.JsonOneLine(singleConfig)] = yml
+	}
+
+	var errMsgs []string
+	for _, snippetConfig := range sourceSnippetConfigs {
+		yml, ok := yamlResults[jsonparse.JsonOneLine(snippetConfig)]
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/snippet.go:65]: please test whether jsonparse.JsonOneLine will print the same result while config has map field.

if not same, please change map to character-ordered string slice and then json marshal.

File: modules/pipeline/services/pipelinesvc/create.go [hunk 3]
```diff
@@ -330,20 +331,17 @@ func (s *PipelineSvc) createPipelineGraph(p *spec.Pipeline, passedDataOpt ...pas
 
 	// 统一处理嵌套流水线
 	// 批量查询 snippet yaml
-	sourceSnippetConfigMap := make(map[string]map[string]apistructs.SnippetConfig) // key: source, value: type
+	var sourceSnippetConfig []apistructs.SnippetConfig
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/create.go:334]: configs

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/services/pipelinesvc/snippet.go (hunk 1 above)
Comment (kakj-go) [reply modules/pipeline/services/pipelinesvc/snippet.go:55]: Whether the subsequent pipeline.yml query of the snippet provider should be a batch query interface, so there is no need to async query here, if async query gittar,query pressure will be too high

File: apistructs/pipeline_yml.go [hunk 4]
```diff
@@ -111,6 +118,36 @@ type SnippetConfig struct {
...
+		for k, v := range snippetConfig.Labels {
+			snippetLabels = append(snippetLabels, SnippetLabel{
+				Key:   k,
+				Value: v,
+			})
+		}
+		sort.Sort(snippetLabels)
+		labels = jsonparse.JsonOneLine(snippetLabels)
+	}
+	return fmt.Sprintf("{\"source\": %s, \"name\": %s, \"labels\": %s}", snippetConfig.Source, snippetConfig.Name, labels)
```
Comment (sfwn) [inline apistructs/pipeline_yml.go:148]: Values of labels will have quote problems.
You can define a anonymous struct and json marshal it.

File: modules/pipeline/services/pipelinesvc/snippet.go [hunk 5]
```diff
@@ -25,81 +26,51 @@ import (
...
-		for _, sc := range singleConfigs {
-			yml, err := s.queryPipelineYAMLBySnippetConfig(&sc)
-			if err != nil {
-				return nil, err
-			}
-			if _, ok := yamlResults[sc.Source]; !ok {
-				yamlResults[sc.Source] = make(map[string]string)
-			}
-			yamlResults[sc.Source][sc.Name] = yml
+	for _, singleConfig := range singleSourceSnippetConfigs {
```
Comment (sfwn) [inline modules/pipeline/services/pipelinesvc/snippet.go:54]: You can handle batch & single at the same time.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Comment by sfwn** [comment]
Please structure a scenario and test time costed before and after this pr.

**Review by sfwn — APPROVED** [review]

**Pull request merged**